plugins/modules/ec2_asg.py

```diff
@@ -220,6 +220,25 @@
       - When I(propagate_at_launch) is true the tags will be propagated to the Instances created.
     type: list
     elements: dict
+  list_tags:
+    description:
+      - If C(true), returns a list of key-value pairs of existing tags on a AutoScalingGroup.
+    type: bool
+    required: false
+    version_added: 3.2.0
+  purge_tags:
+    description:
+      - If C(true), existing tags will be purged from the resource to match exactly what is defined by I(tags) parameter.
+      - If the I(tags) parameter is not set then tags will not be modified.
+    default: true
+    type: bool
+    version_added: 3.2.0
+  remove_tags:
+    description:
+      - If C(true), the tags specified by I(tags) parameter will be removed/deleted from AutoScalingGroup.
+      - If the I(tags) parameter is not set then tags will not be modified.
+    type: bool
+    version_added: 3.2.0
```

Comment:
The question here is: do we want to keep a dedicated to module for doing this or handle tags directly here? @tremble @jillr @markuman If we fall into the handling of tags within this module, the only parameter you need is `purge_tags`. Conversely, if we agree with a new module, you should resurrect the original PR https://github.com/ansible-collections/community.aws/pull/482. 

```suggestion
  purge_tags:
    description:
      - If C(true), existing tags will be purged from the resource to match exactly what is defined by I(tags) parameter.
    default: true
    type: bool
    version_added: 3.2.0
```

Reply:
I don't want a dedicated tags module.  We have ec2_tag and that's fine, but I don't want us to go down the path of a dedicated tag module for every service, either for our own maintenance or for users having to juggle so many modules.  I agree with @alinabuzachis about the diff here as well, CUD modules (like ec2_asg) in amazon.aws should generally not do list-only actions, ec2_asg_info returns tag info for this purpose.  We should just have the same purge_tags behaviour that we do in other, existing modules.
Thanks for your work on this @mandar242!